### PR TITLE
 NFS dependent package installation

### PR DESCRIPTION
**Pull request by abhijeetpatra**
- Check and install nfs-common package
- Changes in config and README

### CODE REVIEW

File: juju-charms/charm-trilio-data-mover/src/reactive/trilio_data_mover.py [hunk 1]
```diff
@@ -91,6 +91,13 @@ def validate_nfs():
     data_dir = config('tv-data-dir')
     device = config('nfs-shares')
 
+    # install nfs-common package
+    cmd = 'apt list --installed'.split()
+    installed_list = check_output(cmd).decode('utf-8')
+    if 'nfs-common' not in installed_list:
```
Comment (MuralidharB) [inline juju-charms/charm-trilio-data-mover/src/reactive/trilio_data_mover.py]: Don't we have a mechanism in charm to claim dependency on nfs-common package?  For example we check nfs-common package in our ansible scripts.
```
- name: install nfs client
  apt:
    pkg: nfs-common
    state: present
  become: true
  when: >
    ansible_distribution_release == 'xenial' and
    backup_target_type == 'NFS'
  tags:
    - configure
```

Comment (abhijeetpatra) [reply juju-charms/charm-trilio-data-mover/src/reactive/trilio_data_mover.py]: Yes, it has a mechanism. The module 'filter_missing_packages' does the exact thing that we require. But it not documented, found this on their repo. Confirmed the usage of this module from James on IRC.

Comment (MuralidharB) [reply juju-charms/charm-trilio-data-mover/src/reactive/trilio_data_mover.py]: Ok, if this is acceptable for Canonical, I am fine with  it.

File: juju-charms/charm-trilio-data-mover/src/reactive/trilio_data_mover.py [hunk 2]
```diff
@@ -91,6 +92,11 @@ def validate_nfs():
     data_dir = config('tv-data-dir')
     device = config('nfs-shares')
 
+    # install nfs-common package
+    if not filter_missing_packages(['nfs-common']):
+        log("'nfs-common' package not found, installing the package...")
+        apt_install(['nfs-common'], fatal=True)
```
[the remark this answers is not in the document]
Comment (abhijeetpatra) [reply juju-charms/charm-trilio-data-mover/src/reactive/trilio_data_mover.py]: We can. Here we are installing the pkg only if the backup-target-type in the config is 'nfs'.
Ideally the dependent packages should be installed at the start of the deployment, irrespective of the config values.

**Pull request merged**